scipy/_lib/_array_api.py

```diff
@@ -651,6 +653,15 @@ def xp_result_device(*args):
     return None
 
 
+# np.r_ replacement
+def concat_1d(xp, *arys):
```

Comment:
```suggestion
def concat_1d(xp: ModuleType | None, *arrays: Iterable[ArrayLike]) -> Array:
```
In line with other functions in this file.